data/data/ibmcloud/variables-ibmcloud.tf

```diff
@@ -110,11 +115,10 @@ variable "ibmcloud_publish_strategy" {
   type = string
   description = "The cluster publishing strategy, either Internal or External"
   default = "External"
-  # TODO: Supported on tf 0.13
-  # validation {
-  #   condition     = "External" || "Internal"
-  #   error_message = "The ibmcloud_publish_strategy value must be \"External\" or \"Internal\"."
-  # }
+  validation {
+    condition = can(regex("^External$|^Internal$", var.ibmcloud_publish_strategy))
+    error_message = "The ibmcloud_publish_strategy value must be \"External\" or \"Internal\"."
+  }
```

Comment:
Yes, that should already be validated since those values are enums
https://github.com/openshift/installer/blob/1bacadd1a8853d56d78e397fbc8e40303283cf47/pkg/types/installconfig.go#L65-L74

If we don't believe this is necessary at the Terraform level, I'll drop this validation check here.

Reply:
Removed, relying on installconfig/validation

Reply:
Not a big deal but I'd say not needed. User input validations should be performed before the terraform stage (in fact, as early as possible).